src/tools/r2rdump/R2RReader.cs

```diff
@@ -77,13 +83,66 @@ public unsafe R2RReader(string filename)
                 }
 
                 DirectoryEntry r2rHeaderDirectory = peReader.PEHeaders.CorHeader.ManagedNativeHeaderDirectory;
-                int r2rHeaderOffset = r2rHeaderDirectory.RelativeVirtualAddress - textSection.VirtualAddress + textSection.PointerToRawData;
-                R2RHeader = new R2RHeader(_image, (uint)r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
+                int r2rHeaderOffset = getOffset(r2rHeaderDirectory.RelativeVirtualAddress, textSection);
+                R2RHeader = new R2RHeader(_image, r2rHeaderDirectory.RelativeVirtualAddress, r2rHeaderOffset);
                 if (r2rHeaderDirectory.Size != R2RHeader.Size)
                 {
                     throw new System.BadImageFormatException("The calculated size of the R2RHeader doesn't match the size saved in the ManagedNativeHeaderDirectory");
                 }
+
+                R2RSection runtimeFunctions = R2RHeader.Sections[R2RSection.SectionType.READYTORUN_SECTION_RUNTIME_FUNCTIONS];
+                int curOffset = getOffset((int)runtimeFunctions.RelativeVirtualAddress, textSection);
+                int nNativeCode = runtimeFunctions.Size / (3*sizeof(int));
+                NativeCode = new NativeCode[nNativeCode];
+                for (int i=0; i<nNativeCode; i++)
+                {
+                    int nativeCodeStartRva = (int)GetField(_image, ref curOffset, sizeof(int));
```

Comment:
Add a helper method for these so that this can be just `int nativeCodeStartRva = GetInt32Field(ref curOffset);` ?